File: UI/eParking_desktop/lib/models/login_response.dart
class LoginResponse {
  final int id;
  final String username;
  final String firstName;
  final String lastName;
  final String email;
  final bool isAdmin;
  final bool isUser;
  final String token;

  LoginResponse({
    required this.id,
    required this.username,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.isAdmin,
    required this.isUser,
    required this.token,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      id: json['id'] as int,
      username: json['username'] as String,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      email: json['email'] as String,
      isAdmin: json['isAdmin'] as bool,
      isUser: json['isUser'] as bool,
      token: json['token'] as String? ?? '',
    );
  }
}

File: eParking.Model/Responses/LoginResponse.cs
namespace eParking.Model.Responses
{
    public class LoginResponse
    {
        public int Id { get; set; }
        public string Username { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public bool IsAdmin { get; set; }
        public bool IsUser { get; set; }
        public string Token { get; set; } = string.Empty;
        public DateTime ExpiresAt { get; set; }
        public string? Picture { get; set; }
    }
}
